### PR TITLE
Add a GitHub Actions workflow to build individual indicators as container images

### PR DESCRIPTION
**Pull request by korlaxxalrok**
### Description

This adds a GitHub Actions workflow to build container images from individual
indicators.

There will be two necesities to invoke the action:

- Create a Dockerfile in the root of the indicator's directory.
- Add the indicator directory name to the `matrix.packages` section of the
  `.github/workflows/build-container-images.yml` like:

  ```
  ...
  jobs:
    build:
      runs-on: ubuntu-latest
      strategy:
        matrix:
          packages: [new_indicator1, new_indicator2] # indicator directory name
  ...
  ```

If the Action succeeds it will build and register a container image in our private
registry. Currently, we will build off of `main` and `prod` branches.

* `main` builds create a registered image of:

  ```text
  ghcr.io/cmu-delphi/covidcast-indicators-${indicator_name}:dev
  ```

* `prod` builds create a registered image of:

  ```text
  ghcr.io/cmu-delphi/covidcast-indicators-${indicator_name}:latest
  ```

### Changelog

Itemize code/test/documentation changes and files added/removed.
- Add a new workflow of `.github/workflows/build-container-images.yml`.
- Update `.github/CONTRIBUTING.md` with basic container image building necessities.
- BONUS! Snuck in some markdown linting changes to `.github/CONTRIBUTING.md`.

### CODE REVIEW

File: .github/workflows/build-container-images.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: cmu-delphi-deploy-machine
+          password: ${{ secrets.CMU_DELPHI_DEPLOY_MACHINE_PAT }}
+
+      - name: Build, tag, and push image to Github
+        env:
```
Comment (sgratzl) [inline .github/workflows/build-container-images.yml:25]: an alternative is changing the working directory directly before the run. Then you don't need docker file specification and the changing directory afterwards

**Comment by sgratzl** [comment]
do you have a template indicator and how the Dockerfile should look like?

**Comment by korlaxxalrok** [comment]
@sgratzl I don't know what that will need to look like just yet. Initially this is so @nmdefries and I can work in the context of containerizing the surveys indicator so we can run it in Automation on Midas. After that we should be able to add a template and enhanced documentation with some practical suggestions.

**Comment by korlaxxalrok** [comment]
I'm going to pull this for now and work with @nmdefries' Dockerfile to test a bit more. I'll submit another PR with an attempt to simplify the workflow (per @sgratzl's suggestion), as well as to provide for something akin to a template indicator.

**Pull request closed without merging**